docs/developer/release-cycle.md

```diff
@@ -0,0 +1,69 @@
+# Release Cycle
+
+The release cycle is the time between when we start working on a release on the `main` branch until the `.0` release (e.g. `v1.3.0`).
+
+**Note**: For the `v1.3` and `v1.4` release cycles we assume a ~ 4 months (~ 17 weeks) release cycle. This should be revisited after the `v1.4.0` release.
+
+The release cycle can be split up into the following phases:
+
+* Week 1-12: Active development
+    * Alpha releases based on the main branch can be released when necessary
```

Comment:
>can be released when necessary

can we define/specify "necessary"?

Reply:
Good question. 

My current point of view is if someone comes up with a reason why they need an alpha release (e.g. it's easier to start testing against a first alpha release of Cluster API) / if someone asks for it.

We could instead also considering implementing nightly releases for release branches. It shouldn't be a lot of effort as we already have them for the main branch.

Those are not full releases but we publish images and manifests and they can be consumed relatively easily (we would have to document them better then we document our nightly releases today though)

Reply:
+1 for nightly releases

Reply:
I will keep alpha in the radar, and we can define if necessary as determined by the release lead.

WRT to nightly let's open an issue and make an assessment if they are really required first (recently generation of nightly release was broken for a couple of weeks and no one complained about it)

Reply:
> recently generation of nightly release was broken for a couple of weeks and no one complained about it

+1. Although I would argue that nightly releases on release branches are more useful. They can be used to verify/use fixes recently merged to release branches of which were using the corresponding releases. I guess nobody is using main and thus nightly release for main are not that useful.

Reply:
> We could instead also considering implementing nightly releases for release branches. It shouldn't be a lot of effort as we already have them for the main branch.

We can't do that until the release branch actually exists which brings us into the RC timeframe... alpha and beta releases would be cut directly from the main branch

Reply:
Good point, thx! I mixed up some things.

But given we already have nightly releases for main, we already have nightly releases which can be used. We should definitely document better how to use them 

They should definitely make alpha releases less necessary :)

That being said, I would keep the current language that the release lead can decide to cut one. Might be useful even if it's just to the new release tooling. We can change it later if it becomes a problem

Reply:
Added a comment to emphasize that for week 1-12 we already have nightly releases.

I'll open a separate issue later to discuss if we want nightly releases for release branches (but they don't matter for the discussion about alpha releases)